docs/source/examples/ddd_tutorial.myst

```diff
@@ -0,0 +1,285 @@
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+# Digital dynamical decoupling (DDD) with Mirror Circuits
+
+In this notebook DDD is applied to improve the success rate of the computation. 
+In DDD, sequences of gates are applied to slack windows, i.e. single-qubit idle windows, in a quantum circuit. 
+Applying such sequences can reduce the coupling between the qubits and the environment, mitigating the effects of noise. 
+For more information on DDD, see the section [DDD section of the user guide](../guide/ddd.myst).
+
+## Setup
+
+We begin by importing the relevant modules and libraries that we will require
+for the rest of this tutorial.
+
+```{code-cell} ipython3
+import functools
+from typing import List, Tuple, Union
+
+# Plotting imports.
+import matplotlib.pyplot as plt
+
+plt.rcParams.update({"font.family": "serif", "font.size": 15})
+%matplotlib inline
+
+# Third-party imports.
+import cirq
+
+import networkx as nx
+import numpy as np
+
+# Mitiq imports.
+from mitiq import benchmarks, ddd
+```
+
+## Define parameters
+
+```{code-cell} ipython3
+# Random seed for circuit generation.
+seed: int = 1
+
+# Total number of shots to use.
+shots: int = 1000
+
+# Qubits to use on the experiment.
+qubits: List[int] = [0, 1, 2]
+
+# Average results over this many trials (circuit instances) at each depth.
+trials: int = 3
+
+# Clifford depths.
+depths: List[int] = [1, 5, 9]
+```
+
+We also define a graph representation of our qubits and assume a line topology.
+
+```{code-cell} ipython3
+# Assume chain-like connectivity
+computer = nx.Graph()
+computer.add_edges_from([(0, 1), (1, 2)])
+nx.draw(computer, with_labels=True)
+
+# Add reversed edges to computer graph.
+# This is important to represent CNOT gates with target and control reversed.
+computer = nx.to_directed(computer)
+```
+
+## Define the circuit
+
+We use mirror circuits to benchmark the performance of the device.
+Mirror circuits, introduced in
+*Proctor et al. arXiv (2021)* {cite}`Proctor_2021_NatPhys` ([arXiv:2008.11294](https://arxiv.org/abs/2008.11294)),
+are designed such that only one bitstring should be sampled. 
+When run on a device, any other measured bitstrings are due to noise.
+The frequency of the correct bitstring is our target metric.
+
+> Note: Mirror circuits build on Loschmidt echo circuits - i.e., circuits of the form $U U^\dagger$ for some unitary $U$.
+Loschmidt echo circuits are good benchmarks but have shortcomings - e.g., they are unable to detect coherent errors.
+Mirror circuits add new features to account for these shortcomings. 
+For more background, see [arXiv:2008.11294](https://arxiv.org/abs/2008.11294).
+
+To define a mirror circuit, we need the device graph. 
+We will use a subgraph of the device, and our first step is picking a subgraph with good qubits.
+
+
+### Generate mirror circuit
+
+Now that we have the device (sub)graph, we can generate a mirror circuit and the bitstring it should sample as follows.
```

Comment:
I'm not sure I follow this stuff about a subgraph, especially since there is no selection of nodes to build a subgraph from.

Reply:
Another copy/paste from PEC mirror circuits. Can we just call it "device graph" and avoid the "subgraph" term?

Reply:
Another copy/paste from PEC mirror circuits. Can we just call it "device graph" and avoid the "subgraph" term?

Reply:
yeah that sounds good.